pkg/healthcheck/healthcheck.go

```diff
@@ -1787,6 +1800,35 @@ func (hc *HealthChecker) checkCanGet(namespace, group, version, resource string)
 	return hc.checkCanPerformAction("get", namespace, group, version, resource)
 }
 
+func (hc *HealthChecker) checkAPIService(serviceName string) error {
+	apiServiceClient, err := apiregistrationv1client.NewForConfig(hc.kubeAPI.Config)
+	if err != nil {
+		return err
+	}
+
+	apiStatus, err := apiServiceClient.APIServices().Get(serviceName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	available := false
+	var errorMessage string
+	for _, condition := range apiStatus.Status.Conditions {
+		if condition.Type == "Available" {
+			available = true
+			if condition.Status == "True" {
```

Comment:
naming this `available` implies that the service is available, but as we can see it might not be. So why noy something like `availableStatusFound` or smth like that ?